egs/wsj/s5/steps/libs/nnet3/train/common.py

```diff
@@ -40,6 +40,57 @@ def __init__(self):
         self.prior_queue_opt = None
         self.parallel_train_opts = None
 
+def get_outputs_list(model_file, get_raw_nnet_from_am=True):
+    """ Generates list of output-node-name used in nnet3 model configuration.
+    """
+    outputs_list=""
+    if get_raw_nnet_from_am:
+      outputs_list, error = common_lib.run_kaldi_command(
+          "nnet3-am-info --print-args=false {0} | "
+          "grep -e 'output-node' | cut -f2 -d' ' | cut -f2 -d'=' ".format(model_file))
+    else:
+      outputs_list, error = common_lib.run_kaldi_command(
+          "nnet3-info --print-args=false {0} | "
+          "grep -e 'output-node' | cut -f2 -d' ' | cut -f2 -d'=' ".format(model_file))
+
+    return outputs_list
+
+def get_multitask_egs_opts(egs_dir, egs_prefix="",
+                           archive_index=1,
+                           use_multitask_egs=False):
+    """ Generates egs option for multitask(or multilingual) training setup,
+        if ouput.scp or weight.scp files exists in egs_dir.
+        Each eg in egs.scp has corresponding task or language in output.scp and
+        weights in weight.scp for scaling supervision for this egs.
+        i.e. egs_prefix is "" for train and
+        "valid_diagnostic." for validation.
+    """
+    multitask_egs_opts=""
+    if use_multitask_egs:
+        output_rename_opt = ""
+        output_file_name="{egs_dir}/{egs_prefix}output.{archive_index}".format(
```

Comment:
Hm, well they are both archive files and text files, but I'd probably lean towards .ark as it will make it more obvious how to supply them as options to programs.